Modules/Challenges/src/Challenges.ConsumerApi/ChallengesModule.cs

```diff
@@ -29,7 +29,10 @@ public override void ConfigureServices(IServiceCollection services, IConfigurati
             dbOptions.DbConnectionString = parsedConfiguration.Infrastructure.SqlDatabase.ConnectionString;
         });
 
-        services.AddSqlDatabaseHealthCheck(Name, parsedConfiguration.Infrastructure.SqlDatabase.Provider, parsedConfiguration.Infrastructure.SqlDatabase.ConnectionString);
+        services.AddSqlDatabaseHealthCheck(Name,
+            parsedConfiguration.Infrastructure.SqlDatabase.Provider,
+            parsedConfiguration.Infrastructure.SqlDatabase.ConnectionString,
+            parsedConfiguration.Infrastructure.SqlDatabase.EnableHealthCheck);
```

Comment:
I don't like that we call `AddSqlDatabaseHealthCheck`, pass `EnableHealthCheck`, and let the method decide whether it ACTUALLY enables add the sql database health check or not. That's not intuitive at all.
Instead, we should probably perform the check outside of the method. 
Same for all occurrences, of course.

Reply:
done